xgboost_ray/main.py

```diff
@@ -219,21 +237,43 @@ def __init__(self,
                  rank: int,
                  num_actors: int,
                  queue: Optional[Queue] = None,
+                 stop_event: Optional[Event] = None,
                  checkpoint_frequency: int = 5):
         self.queue = queue
-        init_session(rank, queue)
+        init_session(rank, self.queue)
 
         self.rank = rank
         self.num_actors = num_actors
 
         self.checkpoint_frequency = checkpoint_frequency
 
         self._data: Dict[RayDMatrix, xgb.DMatrix] = {}
-
         self._local_n = 0
 
+        self._stop_event = stop_event
+
         _set_omp_num_threads()
 
+    def set_queue(self, queue: Queue):
+        self.queue = queue
+        set_session_queue(self.queue)
+
+    def set_stop_event(self, stop_event: Event):
+        self._stop_event = stop_event
+
+    def pid(self):
+        """Get process PID. Used for checking if still alive"""
+        return os.getpid()
+
+    def _stop_training_callback(self):
```

Comment:
It stops training if the event is set. `asyncio.Event` is just the internal structure the Event ray actor uses (just like the Queue ray actor uses `asyncio.Queue`. However I don't think we need this callback anymore since callbacks aren't processed anymore once a local training actor died. I'll remove it